edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs

```diff
@@ -44,6 +46,7 @@ public class ConnectionManager : IConnectionManager
             this.credentialsCache = Preconditions.CheckNotNull(credentialsCache, nameof(credentialsCache));
             this.edgeDeviceId = Preconditions.CheckNonWhiteSpace(edgeDeviceId, nameof(edgeDeviceId));
             this.edgeModuleId = Preconditions.CheckNonWhiteSpace(edgeModuleId, nameof(edgeModuleId));
+            Util.Metrics.RegisterGaugeCallback(() => Metrics.SetConnectedClientCountGauge(this.GetConnectedClients().Count()));
```

Comment:
Nit: Why not Metrics.RegisterGaugeCallback(() => this.GetConnectedClients().Count())) to keep this clean?

Reply:
Just to confirm - This will not be called if Metrics are not enabled, right?

Reply:
Correct. Cannot call it directly like you suggested because some options need to be passed in.